django_mongodb/base.py

```diff
@@ -157,20 +159,35 @@ def __getattr__(self, attr):
         raise AttributeError(attr)
 
     def _connect(self):
-        settings_dict = self.settings_dict
+        settings_dict = deepcopy(self.settings_dict)
+        user = settings_dict.get("USER")
+        password = settings_dict.get("PASSWORD")
+        options = settings_dict.get("OPTIONS", {})
+        options_user = options.pop("username", None)
+        options_password = options.pop("password", None)
+        # Ensure that the user did not specify both USER and username in OPTIONS.
+        if user and options_user:
+            raise ImproperlyConfigured(
+                f"In the {self.alias} settings, both USER and OPTIONS['username'] are set."
+            )
+        if password and options_password:
+            raise ImproperlyConfigured(
+                f"In the {self.alias} settings, both PASSWORD and OPTIONS['password'] are set."
+            )
         self.connection = MongoClient(
             host=settings_dict["HOST"] or None,
             port=int(settings_dict["PORT"] or 27017),
-            **settings_dict["OPTIONS"],
+            username=user or options_user,
+            password=password or options_password,
+            **options,
         )
         db_name = settings_dict["NAME"]
         if db_name:
             self.database = self.connection[db_name]
 
-        user = settings_dict["USER"]
-        password = settings_dict["PASSWORD"]
-        if user and password and not self.database.authenticate(user, password):
-            raise ImproperlyConfigured("Invalid username or password.")
+        # We can check if the credentials are correct by calling server_info().
```

Comment:
Thanks for the explanation, but I think we can omit the comment since this is no different from the behavior of other backends.